pruning/keep_grid.go

```diff
@@ -15,42 +14,31 @@ import (
 // uses the most recent snapshot among those that match the regex as 'now',
 // and deletes all snapshots that do not fit the grid specification.
 type KeepGrid struct {
+	KeepCommon
 	retentionGrid *retentiongrid.Grid
-	re            *regexp.Regexp
 }
 
-func NewKeepGrid(in *config.PruneGrid) (p *KeepGrid, err error) {
-
+func NewKeepGrid(in *config.PruneGrid) (*KeepGrid, error) {
 	if in.Regex == "" {
 		return nil, fmt.Errorf("Regex must not be empty")
 	}
-	re, err := regexp.Compile(in.Regex)
+
+	kc, err := newKeepCommon(&in.PruneKeepCommon)
 	if err != nil {
-		return nil, errors.Wrap(err, "Regex is invalid")
+		return nil, err
 	}
 
-	return newKeepGrid(re, in.Grid)
+	return newKeepGrid(kc, in.Grid)
 }
 
-func MustNewKeepGrid(regex, gridspec string) *KeepGrid {
+func newKeepGrid(common KeepCommon, configIntervals []config.RetentionInterval) (*KeepGrid, error) {
 
-	ris, err := config.ParseRetentionIntervalSpec(gridspec)
-	if err != nil {
-		panic(err)
+	if common.regex == nil {
+		panic("regex must not be nil")
 	}
 
-	re := regexp.MustCompile(regex)
-
-	grid, err := newKeepGrid(re, ris)
-	if err != nil {
-		panic(err)
-	}
-	return grid
-}
-
-func newKeepGrid(re *regexp.Regexp, configIntervals []config.RetentionInterval) (*KeepGrid, error) {
-	if re == nil {
-		panic("re must not be nil")
+	if common.filesystems == nil {
```

Comment:
Why do we have this test for `keep_grid.go`, but not the other keep rules?
Shouldn't `newKeepCommon`, in combination with the default value in `config`, guarantee that we always have a non-nil filesystems filte?

Reply:
You're right, good catch!